lib/charms/opensearch/v0/opensearch_snapshots.py

```diff
@@ -1694,3 +1469,78 @@ def ca_changed(self, new_chain: str | None) -> bool:
         )
         logger.debug("The old and new CA chains are different: %s", is_different)
         return is_different
+
+    def update_ca(self, ca: str | None) -> None:
+        """Adds or removes custom S3 ca"""
+        if ca:
+            logger.info("S3 TLS CA Chain detected.")
+            ca_changed = self.ca_changed(ca)
+            self.store_s3_ca(ca)
+            if ca_changed:
+                self.charm.request_opensearch_restart(reason="apply object storage CA change")
+        elif self.is_custom_s3_ca_stored():
+            # If we had a custom CA but peer no longer provides one, clean it up
+            self.remove_s3_ca()
+            self.charm.request_opensearch_restart(reason="clean up the object storage CA")
+
+    def get_backup_label(self, object_storage_type):
+        """Returns plugin label associated with object storage type"""
+        if object_storage_type == ObjectStorageType.S3:
+            return S3_LABEL
+        if object_storage_type == ObjectStorageType.AZURE:
+            return AZURE_LABEL
+
+    def ensure_snapshot_secrets(self):
+        """Checks if backup relation exists without entry in stored plugins"""
+        # if this charm was upgraded and already had a backup relation,
+        # the credentials will not exist yet in this charm's stored plugin_config_info.
+        # This handler creates and stores the secret containing the credentials
+        if not self.charm.unit.is_leader():
+            return
+
+        if self.charm.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR:
+            return
+
+        # check if a backup relation exists with no secret
+        if not (object_storage_config := self.get_storage_config()):
+            return
+
+        object_storage_type = self.get_storage_type()
+        if not (label := self.get_backup_label(object_storage_type)):
+            logger.warning(
+                "Could not determine label for object storage type: %s", object_storage_type
+            )
+            return
+
+        if (
+            object_storage_config := self.get_storage_config()
```

Comment:
```suggestion
            object_storage_config
```
We already defined object_storage_config.